http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/websocket/message/WebSocketInitMessageImpl.java

```diff
@@ -56,49 +57,61 @@ public WebSocketInitMessageImpl(ChannelHandlerContext ctx, HttpRequest httpReque
 
     @Override
     public Session handshake() throws ProtocolException {
-
         WebSocketServerHandshakerFactory wsFactory =
-                new WebSocketServerHandshakerFactory(getWebSocketURL(httpRequest), getSubProtocol(), true);
+                new WebSocketServerHandshakerFactory(getWebSocketURL(httpRequest), null, true);
         WebSocketServerHandshaker handshaker = wsFactory.newHandshaker(httpRequest);
-        try {
-            handshaker.handshake(ctx.channel(), httpRequest);
-
-            //Replace HTTP handlers  with  new Handlers for WebSocket in the pipeline
-            ChannelPipeline pipeline = ctx.pipeline();
-            pipeline.addLast(Constants.WEBSOCKET_SOURCE_HANDLER, webSocketSourceHandler);
+        return handleHandshake(handshaker, 0);
+    }
 
-            pipeline.remove(Constants.IDLE_STATE_HANDLER);
-            pipeline.remove(Constants.HTTP_SOURCE_HANDLER);
+    @Override
+    public Session handshake(String[] subProtocols, boolean allowExtensions) throws ProtocolException {
+        WebSocketServerHandshakerFactory wsFactory =
+                new WebSocketServerHandshakerFactory(getWebSocketURL(httpRequest), getSubProtocolsStr(subProtocols),
+                                                     allowExtensions);
+        WebSocketServerHandshaker handshaker = wsFactory.newHandshaker(httpRequest);
```

Comment:
newHandshaker sounds a like noun rather than a verb. But methods represents actions therefore shouldn't it better to have a verb ?

Reply:
This method is coming from netty itself. We have just used it.